libp2p/Common.h

```diff
@@ -254,26 +254,24 @@ class DeadlineOps
     public:
         DeadlineOp(ba::io_service& _io, unsigned _msInFuture, std::function<void(boost::system::error_code const&)> const& _f): m_timer(new ba::deadline_timer(_io)) { m_timer->expires_from_now(boost::posix_time::milliseconds(_msInFuture)); m_timer->async_wait(_f); }
         ~DeadlineOp() { if (m_timer) m_timer->cancel(); }
-        
-        DeadlineOp(DeadlineOp&& _s): m_timer(_s.m_timer.release()) {}
-        DeadlineOp& operator=(DeadlineOp&& _s)
-        {
-            assert(&_s != this);
 
-            m_timer.reset(_s.m_timer.release());
+        DeadlineOp(DeadlineOp&& _s) noexcept : m_timer(std::move(_s.m_timer)) {}
+        DeadlineOp& operator=(DeadlineOp&& _s) noexcept
+        {
+            m_timer = std::move(_s.m_timer);
             return *this;
         }
-        
+
         bool expired() { Guard l(x_timer); return m_timer->expires_from_now().total_nanoseconds() <= 0; }
         void wait() { Guard l(x_timer); m_timer->wait(); }
-        
+
     private:
         std::unique_ptr<ba::deadline_timer> m_timer;
         Mutex x_timer;
     };
-    
+
 public:
-    DeadlineOps(ba::io_service& _io, unsigned _reapIntervalMs = 100): m_io(_io), m_reapIntervalMs(_reapIntervalMs), m_stopped(false) { reap(); }
+    explicit DeadlineOps(ba::io_service& _io, unsigned _reapIntervalMs = 100): m_io(_io), m_reapIntervalMs(_reapIntervalMs), m_stopped(false) { reap(); }
```

Comment:
Just to verify I understand this correctly, is the benefit of "explicit" here that we get a more informative compile-time error if the default constructor is invoked (since without explicit I think invoking the default constructor would trigger a compiler error anyway since DeadlineOps holds a reference to the IO service which needs to be initialized)

Reply:
Yes, this blocks implicit conversion from `io_service`.